diff --git a/dwm.c b/dwm.c
index 99a701b..a7c6e9c 100644
--- a/dwm.c
+++ b/dwm.c
@@ -833,10 +833,12 @@ drawbar(Monitor *m)
 		drw_text(drw, x, 0, w, bh, lrpad / 2, tags[i], urg & 1 << i);
 		if (ulineall || m->tagset[m->seltags] & 1 << i) /* if there are conflicts, just move these lines directly underneath both 'drw_setscheme' and 'drw_text' :) */
 			drw_rect(drw, x + ulinepad, bh - ulinestroke - ulinevoffset, w - (ulinepad * 2), ulinestroke, 1, 0);
-		if (occ & 1 << i)
+			// Draw little square only on non selected tags
+		if (occ & 1 << i && ! (m->tagset[m->seltags] & 1 << i )){
 			drw_rect(drw, x + boxs, boxs, boxw, boxw,
 				m == selmon && selmon->sel && selmon->sel->tags & 1 << i,
 				urg & 1 << i);
+		}
 		x += w;
 	}
 	w = TEXTW(m->ltsymbol);
